src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -105,6 +129,29 @@ internal sealed class CSharpChangeSignatureService : AbstractChangeSignatureServ
             return symbolInfo.Symbol ?? symbolInfo.CandidateSymbols.FirstOrDefault();
         }
 
+        private SyntaxNode GetMatchingNode(SyntaxNode node, bool restrictToDeclarations)
+        {
+            var matchKinds = restrictToDeclarations
+                ? _declarationKinds
+                : _declarationAndInvocableKinds;
+
+            for (var current = node; current != null; current = current.Parent)
+            {
+                if (restrictToDeclarations &&
+                    current.Kind() == SyntaxKind.Block || current.Kind() == SyntaxKind.ArrowExpressionClause)
```

Comment:
> why

Because if you say that you only want to be on a declaration, but you hit a { } or => on your way up, then you're definitely not on a declaration.

> Does this do the right thing for local functions?

It looks like there's no support for change-sig for local functions, period.  :)  (But, if there were, this logic would be sound).